File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/screen/main/users/UserPopularAdapter.java
package ru.avystavkin.githubgists.screen.main.users;


import android.view.LayoutInflater;
import android.view.ViewGroup;

import java.util.List;

import androidx.annotation.NonNull;
import ru.avystavkin.githubgists.R;
import ru.avystavkin.githubgists.content.User;
import ru.avystavkin.githubgists.screen.base.adapters.BaseAdapter;


public class UserPopularAdapter extends BaseAdapter<UserViewHolder, User> {

    public UserPopularAdapter(@NonNull List<User> items) {
        super(items);
    }

    @Override
    public UserViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new UserViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_user, parent, false));
    }

    @Override
    public void onBindViewHolder(UserViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);
        holder.bind(getItem(position));
    }
}

File: GithubGists/app/src/main/java/ru/avystavkin/githubgists/screen/main/users/UserPopularAdapter.kt
package ru.avystavkin.githubgists.screen.main.users


import android.view.LayoutInflater
import android.view.ViewGroup
import ru.avystavkin.githubgists.R
import ru.avystavkin.githubgists.models.database.User
import ru.avystavkin.githubgists.screen.base.adapters.BaseAdapter


class UserPopularAdapter(items: List<User>) : BaseAdapter<UserViewHolder, User>(items) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        return UserViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.item_user, parent, false))
    }

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)
        holder.bind(getItem(position))
    }
}

